### ISSUE TITLE
Monotorrent fails to gracefully handle absolute paths in torrent files

### ISSUE THREAD

**Issue by issueg2k4g34j2g** [alanmcgovern/monotorrent#517]
## Expected Behavior
monotorrent should silently rewrite absolute paths into relative paths instead of throwing an error message.

## Current Behavior
LoadTorrentFiles chokes when supplied with a torrent file containing absolute paths.

The responsible code can be found at https://github.com/alanmcgovern/monotorrent/blob/999d34b5f102aa4dfb72497b381e8f36b86f5653/src/MonoTorrent.Client/MonoTorrent/PathValidator.cs#L38-L46

and

https://github.com/alanmcgovern/monotorrent/blob/b0c484684c708bfee4d92c06178f1e64d2cc4399/src/MonoTorrent.Client/MonoTorrent/Torrent.cs#L646-L655

## Possible Solution
All common torrent clients deal with this gracefully by rewriting absolute paths into relative paths. 

A simple solution would be to simply disregard leading empty path segments (and slashes?) while parsing the paths in LoadTorrentFilesV1. 

This would also be a good solution from a security POV, as it would not require modifications to the path validation code.

## Steps to Reproduce
1.  Attempt to load this torrent file with monotorrent
[Fargo.S01.1080p.BluRay.DTS5.1.x264-ROVERS.[BTN].torrent.zip](https://github.com/alanmcgovern/monotorrent/files/8311854/Fargo.S01.1080p.BluRay.DTS5.1.x264-ROVERS.BTN.torrent.zip)

## Context (Environment)
I have run into several torrent files demonstrating this issue while diagnosing issues with my sonarr setup, they appear to be fairly common in the wild.
Most of these files appear to originate from ruTorrent, which seems to be a fairly common software used to produce torrent files. Other software (deluge, transmission, rtorrent, qbittorrent, etc...) handle these files gracefully, so they are not treated as corrupt by various indexers.

These files appear to be fully spec compliant, just strange.

**Comment by issueg2k4g34j2g** [comment]
My apologies for not including a pull request despite this being such a simple issue, I'm completely out of my depth with C#.

**Comment by issueg2k4g34j2g** [comment]
So I figured I'd try to make a useful pull request anyway, this *should work* as far as I can tell.

It is worth considering whether or not this should be only limited to leading directory separators, but I can't imagine a legitimate use case for having extra directory separators anywhere in the path.

**Comment by alanmcgovern** [comment]
Your analysis was *near* perfect!

I took a quick look at the actual data contained within the file you specified and it looks like this:
```
"path": [
               "",
               "fargo.s01e07.1080p.bluray.x264-rovers.mkv"
            ]
```
The issue is that monotorrent is accidentally running the equivalent of: `"" + Path.DirectorySeparatorChar, "fargo.s01e07.1080p.bluray.x264-rovers.mkv"` as the first part of the array is empty. The generated path is an absolute path, which is why it ends up being rejected later on.

I compared the behaviour of qBittorrent and uTorrent and they both handle this differently. qBittorrent replaces the empty string with `_` and uTorrent discards it.

In qBitorrent's case all the files end up in an unexpected subdirectory, but it's not a bad outcome as such. A small tweak to your patch which discards empty strings may be sufficient to resolve this!
